Update map.component.ts

pierre56 opened the pull request:
(L.control as any).locate({
          position: 'topright',    // place le bouton en haut à droite
          watch: true,             // active le suivi permanent (navigator.watchPosition)
          setView: 'always',       // recentre la carte à chaque mise à jour
          flyTo: true,             // animation fluide pour les transitions
          showPopup: false         // pas de popup "Vous êtes ici"

  

Modif afin d'avoir l'option de géoloc disponible même sans activer l'édition pour monitoring,
 en plaçant le bouton en haut a droite.
 
 Ajout des options watch -> tracking pour suivre le déplacement
 et setview -> utile pour revenir automatiquement a la position gps, même en changeant de fenêtre
 
 exemple concret : 
 Je finis de saisir une observation sur un site, je reviens au niveau module. 
 Je suis recentré automatiquement sur ma nouvelle position GPS proche de mon site suivant, ce qui permet de le sélectionner facilement via la carto (pas besoin de taper le nom dans le filtre).
 
<img width="952" height="487" alt="image" src="https://github.com/user-attachments/assets/2fafb8c9-f3aa-4efd-83b9-5a8da03f62c1" />

pierre56 committed:
Update map.component.ts


       (L.control as any).locate({
          position: 'topright',    // place le bouton en haut à droite
          watch: true,             // active le suivi permanent (navigator.watchPosition)
          setView: 'always',       // recentre la carte à chaque mise à jour
          flyTo: true,             // animation fluide pour les transitions
          showPopup: false         // pas de popup "Vous êtes ici"

  

Modif afin d'avoir l'option de géoloc disponible même sans activer l'édition en plaçant le bouton en haut a droite.
utile pour monitoring

jacquesfize reviewed (commented):
Ok pour le déplacement de l'icône, c'est plus cohérent en terme d'usages.

Pour moi le paramètre `flyTo` doit être mis à `false`, car ralentit l'utilisateur dans son usage de la geoloc dans GN.

Pour le absence de l'icône de geoloc dans la carte de monitorings, j'ai ouvert une issue https://github.com/PnX-SI/gn_module_monitoring/issues/505.

pierre56 committed:
Update frontend/src/app/GN2CommonModule/map/map.component.ts

Lors de mes tests, j'ai pas eu trop de lags.
Je trouvais l'animation sympa mais pas de soucis pour que ça dégage :)

Co-authored-by: Jacques Fize <4259846+jacquesfize@users.noreply.github.com>

camillemonchicourt commented:
C'est normal que le bouton n'était pas visible actuellement en mode LISTE, il était uniquement prévu quand on est en saisie d'une observation pour faciliter le fait de se localiser facilement sur sa position si on utilise GeoNature sur le terrain (qui n'est pas prévu pour ça comme déjà évoqué).
Mais OK, pourquoi pas ajouter le bouton aussi sur les carte en mode Consultation / Listes, mais que si on active la fonctionnalité de localisation GPS, qui ne l'est pas par défaut, car on ne peut pas laisser penser actuellement que GeoNature est adaptée à un usage mobile, car cela ne fait pas partie de son périmètre global actuellement.

jacquesfize commented:
>  Mais OK, pourquoi pas ajouter le bouton aussi sur les carte en mode Consultation / Listes, mais que si on active la fonctionnalité de localisation GPS, qui ne l'est pas par défaut, car on ne peut pas laisser penser actuellement que GeoNature est adaptée à un usage mobile, car cela ne fait pas partie de son périmètre global actuellement.

Dans la manière dans lequel a été conçu le composant Carte --dans tous les cas-- la géolocalisation est activé uniquement si le paramètre en configuration est indiqué à vrai.

camillemonchicourt commented:
Car je pense que bien faire une application web/mobile est une vraie approche globale, technique et fonctionnelle qui demande de la conception, des fonctionnalités, des cas d'usage bien identifiés, etc...
Et GeoNature a été conçu et développé pour un usage web desktop. Pour le mobile, on s'est concentré sur Occtax-mobile et désormais Monitoring-mobile.

Je ne dis pas que cela ne fonctionne pas avec des ajustements et "bidouilles", mais ce n'est pas prévu globalement pour et on n'a pas la capacité de lancer un tel sujet actuellement et de le supporter.
Mais vu que ça fonctionne à peu prêt pour ceux qui veulent, on a intégré l'ajout de la géolocalisation sans l'activer par défaut, pour ceux qui veulent mais sans assumer/supporter que GeoNature est globalement adapté et fonctionnel en mobile.

camillemonchicourt commented:
A priori, cette PR n'est plus nécessaire depuis les corrections apportées dans Monitoring 1.2.4 : https://github.com/PnX-SI/gn_module_monitoring/releases/tag/1.2.4

jacquesfize commented:
Problème corrigé dans la 1.2.4
